Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1602,48 +1602,53 @@ SaveNvsData (
   if (EFI_ERROR(Status)) {
     return EFI_NOT_FOUND;
   }
-  if (*(UINT32 *)Address == 0xFFFFFFFF) {
-    Status = SpiConstructor ();
-    if (EFI_ERROR (Status)) {
-      return Status;
-    }
 
-    Status = SpiGetRegionAddress (FlashRegionAll, NULL,  &TotalSize);
-    if (EFI_ERROR (Status)) {
-      return Status;
-    }
+  //
+  // Compare input data against the stored MRC training data
+  // if they match, no need to update again.
+  //
+  if (CompareMem ((VOID *)Address, Buffer, Length) == 0){
+    return EFI_ALREADY_STARTED;
+  }
```

Comment:
Late need optimize this check to avoid erasing and saving all these data.

Reply:
Okay